.github/workflows/create-release.yml

```diff
@@ -1,279 +1,456 @@
-          echo "Relevant files have changed since previous release."
-          echo "${changed}"
-          echo "stack_files_changed=true" >> "$GITHUB_OUTPUT"
-        fi
-
-  run_if_stack_files_changed:
-    name: Run If Stack Files Changed
-    runs-on: ubuntu-22.04
-    needs: [stack_files_changed]
-    if: ${{ needs.stack_files_changed.outputs.stack_files_changed == 'true' }}
+      ubi_images_need_update: ${{ steps.compare_previous_and_current_sha256_hash_codes.outputs.ubi_images_need_update }}
+
     steps:
-    - name: Run if stack files changed
-      run: |
-        echo "stack files have changed"
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Fetch previous build sha256 image hash code
+        id: fetch_previous_build_sha256_image_hash_code
+        uses: paketo-buildpacks/github-config/actions/release/find-and-download-asset@main
+        with:
+          asset_pattern: "${{ env.UBI_BUILD_SHA256_FILENAME }}"
+          search_depth: 1
+          repo: ${{ github.repository }}
+          output_path: "/github/workspace/${{ env.UBI_BUILD_SHA256_FILENAME }}"
+          token: ${{ secrets.PAKETO_BOT_GITHUB_TOKEN }}
+
+      - name: Fetch previous run sha256 image hash code
+        id: fetch_previous_run_sha256_image_hash_code
+        uses: paketo-buildpacks/github-config/actions/release/find-and-download-asset@main
+        with:
+          asset_pattern: "${{ env.UBI_RUN_SHA256_FILENAME }}"
+          search_depth: 1
+          repo: ${{ github.repository }}
+          output_path: "/github/workspace/${{ env.UBI_RUN_SHA256_FILENAME }}"
+          token: ${{ secrets.PAKETO_BOT_GITHUB_TOKEN }}
+
+      - name: Fetch previous run Node.js 16 sha256 image hash code
+        id: fetch_previous_run_nodejs_16_sha256_image_hash_code
+        uses: paketo-buildpacks/github-config/actions/release/find-and-download-asset@main
+        with:
+          asset_pattern: "${{ env.UBI_RUN_NODEJS_16_SHA256_FILENAME }}"
+          search_depth: 1
+          repo: ${{ github.repository }}
+          output_path: "/github/workspace/${{ env.UBI_RUN_NODEJS_16_SHA256_FILENAME }}"
+          token: ${{ secrets.PAKETO_BOT_GITHUB_TOKEN }}
+
+      - name: Fetch previous run Node.js 18 sha256 image hash code
+        id: fetch_previous_run_nodejs_18_sha256_image_hash_code
+        uses: paketo-buildpacks/github-config/actions/release/find-and-download-asset@main
+        with:
+          asset_pattern: "${{ env.UBI_RUN_NODEJS_18_SHA256_FILENAME }}"
+          search_depth: 1
+          repo: ${{ github.repository }}
+          output_path: "/github/workspace/${{ env.UBI_RUN_NODEJS_18_SHA256_FILENAME }}"
+          token: ${{ secrets.PAKETO_BOT_GITHUB_TOKEN }}
+
+      - name: Compare previous and current sha256 hash codes
+        id: compare_previous_and_current_sha256_hash_codes
+        run: |
+
+          ubi_images_need_update=()
+
+          # Run Image 
+          docker pull registry.access.redhat.com/ubi8/ubi-minimal
```

Comment:
@pacostas we should not be checking the ubi images, but intead the Node.js images for changes. For each LTS line we need to check if there is a new release, currently the ones we should check would include:

registry.access.redhat.com/ubi8/nodejs-16
registry.access.redhat.com/ubi8/nodejs-18
registry.access.redhat.com/ubi8/nodejs-16-minimal
registry.access.redhat.com/ubi8/nodejs-18-minimal

If there is a new version of any of those images then we should trigger the build of the stack